src/query/api/v1/handler/prometheus/native/common_test.go

```diff
@@ -212,11 +218,109 @@ func TestRenderResultsJSON(t *testing.T) {
 						]
 					],
 					"step_size_ms": 10000
+				},
+				{
+					"metric": {
+						"biz": "baz",
+						"qux": "qaz"
+					},
+					"values": [
+						[
+							1535948880,
+							"NaN"
+						],
+						[
+							1535948890,
+							"NaN"
+						]
+					],
+					"step_size_ms": 10000
 				}
 			]
 		}
 	}
 	`)
+
+	actual := mustPrettyJSON(t, buffer.String())
+	assert.Equal(t, expected, actual, xtest.Diff(expected, actual))
+}
+
+func TestRenderResultsJSONWithDroppedNaNs(t *testing.T) {
+	start := time.Unix(1535948880, 0)
+	buffer := bytes.NewBuffer(nil)
+	params := models.RequestParams{}
+	valsWithNaN := ts.NewFixedStepValues(10*time.Second, 2, 1, start)
+	valsWithNaN.SetValueAt(1, math.NaN())
+
+	series := []*ts.Series{
+		ts.NewSeries([]byte("foo"),
+			valsWithNaN, test.TagSliceToTags([]models.Tag{
+				models.Tag{Name: []byte("bar"), Value: []byte("baz")},
+				models.Tag{Name: []byte("qux"), Value: []byte("qaz")},
+			})),
+		ts.NewSeries([]byte("bar"),
+			ts.NewFixedStepValues(10*time.Second, 2, 2, start), test.TagSliceToTags([]models.Tag{
```

Comment:
Have one of these tests have some nans

Reply:
Actually `TestRenderResultsJSONWithDroppedNaNs` already has a `NaN` and a real number